### DISCUSSION TITLE
ECSQL - INFORMATION_SCHEMA.COLUMNS

### DISCUSSION THREAD

**Discussion by PaoloPetra** [iTwin/itwinjs-core#3296] [Q&A] [answered]
Hello everybody
i'm trying to extract the properties of a class from ecsql

through this query I can figure out which tables I have to query to extract the properties various properties

SELECT aspect.ecclassid
FROM bis.GeometricElement element
JOIN meta.ECClassDef class ON element.ECClassId = class.ECInstanceId
JOIN BisCore.ElementMultiAspect aspect on element.ECInstanceId = aspect.element.id
WHERE class.displaylabel = 'xxxxxxxxxx'
group by aspect.ecclassid

at this point I find that the properties are developed on a line
is there a way in ecsql type INFORMATION_SCHEMA.COLUMNS of SQL to read the columns of those classes ???

I'm doing it wrong ? is there a faster way to determine the properties of a class ??

THANKS FOR THE PRECIOUS HELP
Paul

**Comment by grigasp** [comment, marked as answer]
I'm not sure I understand your question, but it seems you want to get properties information of specific class. That can be achieved through this ECSQL query:
```sql
select p.*
from meta.ecpropertydef p
join meta.classhasallbaseclasses bc on bc.targetecinstanceid = p.class.id
where bc.sourceecinstanceid = {class_id}
```

BTW, the `bis.ElementMultiAspect` class is not the only class that may contain element properties - there's also `bis.ElementUniqueAspect`. Also, Design Review, in addition to direct element properties and its aspect properties, loads some others from grouping element, physical type, etc. If you want your properties list to match what's displayed there, it's recommended to use the [Presentation](https://www.itwinjs.org/presentation/) APIs. If you're interested not only in property information, but also getting the values, you may want to have a look at the [getElementProperties](https://github.com/iTwin/itwinjs-core/blob/7890b9d2434734d7c2b01acb92a4e51d2334727a/presentation/frontend/src/presentation-frontend/PresentationManager.ts#L415) function ([example](https://www.itwinjs.org/sandboxes/grigas/Element%20Properties%20Loader)).

**Reply by PaoloPetra** [reply]
Thanks for the reply grigasp !! :)

I'm trying to do is create a more accurate search similar to the one on https://connect-imodelweb.bentley.com/

given a certain class, extrapolate the various properties in this case 18 properties, and then do a reload for the single property with "=,>, <, like" etc. etc.

I am attaching a photo that should make you understand my purpose

the query I did earlier helps me understand which are the tables of that property but not the properties themselves.


![ifcCovering](https://user-images.githubusercontent.com/40106886/156588636-ba7c44d6-86fd-49cd-a685-1988dc7e20ff.png)

**Reply by ColinKerr** [reply]
I did two videos focused on querying an iModel.  The first video gives an overview of how data is stored in an iModel.  The second video includes a little bit about presentation rules.  You can find the videos here (they are the first two videos in this playlist): https://www.youtube.com/playlist?list=PL6YCKeNfXXd9VFzTF-NSeN_l5vB7OTWR6
